### PR TITLE
Update DYDX rewards formulas for DIP-7

### PR DESCRIPTION
**Pull request by boxhock**
## Changes

- Added types for IPFS data
- Extended the poke() method to support different formulas for the calculations based on the epoch
- Added new formulas for [DIP-7](https://github.com/dydxfoundation/dip/blob/master/content/dips/DIP-7.md)

### Note:

New environment variables:

```dotenv
TRADER_SCORE_A=670000000000000000
TRADER_SCORE_B=280000000000000000
TRADER_SCORE_C=050000000000000000
```

## Steps to Test

1. Run integration tests
2. LinkPool to test on testnet

## Quality Assurance

- [x] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `<ADAPTER_PACKAGE>/schemas/env.json` and `<ADAPTER_PACKAGE>/README.md`
- [ ] If a new adapter was made, or an existing one was modified so that its environment variables have changed, update the relevant `infra-k8s` configuration file.
- [x] The branch naming follows git flow (`feature/x`, `chore/x`, `release/x`, `hotfix/x`, `fix/x`) or is created from Clubhouse/Shortcut
- [x] This is related to a maximum of one Clubhouse/Shortcut story or GitHub issue
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)

### CODE REVIEW

File: packages/composites/dydx-rewards/src/method/formulas/dip-7.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  b = 0.28, // TRADER_SCORE_B=280000000000000000
+  c = 0.05, // TRADER_SCORE_C=050000000000000000
+): void => {
+  const traderScore: { [address: string]: bn.BigNumber } = {}
+  let traderScoreSum = new bn.BigNumber(0)
+
+  Object.keys(epochData.tradeFeesPaid).forEach((addr) => {
+    const f = epochData.tradeFeesPaid[addr]
+    const d = epochData.openInterest[addr] || 0
+    const g = epochData.averageActiveStakedDYDX[addr] || 0
```
Comment (jiajames) [inline packages/composites/dydx-rewards/src/method/formulas/dip-7.ts:19]: I might be missing something here: are these the raw `OracleRewardsData` values? What I noticed is that these values are divided by the `totalTradeFeesPaid F`, `openInterest D` like in your previous formula [here](https://github.com/smartcontractkit/external-adapters-js/blob/7754a9581f6eed8156a697dd474c6c2f67126db3/packages/composites/dydx-rewards/src/method/formulas/initial.ts#L108)

File: packages/composites/dydx-rewards/test/integration/fixtures.ts [hunk 2]
```diff
@@ -549,4 +550,36 @@ export function mockIpfsResponseSuccess(): void {
       'Transfer-Encoding',
       'chunked',
     ])
+    .post('/api/v0/add', fileUploadMatches(JSON.stringify(expectedTestData2)))
+    .query({ 'stream-channels': 'true', 'cid-version': '1', progress: 'false' })
+    .reply(
+      200,
+      {
+        Name: 'bafkreibh6gurutmhxgopygg5oi446rguehhgwgv4xx2jusw3hlzku7qzzm',
```
Comment (austinborn) [inline packages/composites/dydx-rewards/test/integration/fixtures.ts:558]: Is this value sensitive?

File: packages/composites/dydx-rewards/src/method/formulas/initial.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+    volumeRequirement: 5_000,
+  },
+  {
+    min: 1,
+    reward: rewardAmountToBigNumber(1163.51),
+    volumeRequirement: 500,
+  },
+  {
+    min: 0,
+    reward: rewardAmountToBigNumber(310.75),
```
Comment (austinborn) [inline packages/composites/dydx-rewards/src/method/formulas/initial.ts:34]: Where did you pull these numbers from? Can you add a link to them in the comments around line 10?

File: packages/composites/dydx-rewards/src/method/formulas/initial.ts [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+
+  const tier = retroactiveTiers.find(({ min }) => tradeVolume >= min)
+  if (!tier) {
+    throw new Error(`Unable to find tier for volume: ${tradeVolume}`)
+  }
+
+  return tier
+}
+
+const EXPO_BONUS_TOKENS = rewardAmountToBigNumber(565.61)
```
Comment (austinborn) [inline packages/composites/dydx-rewards/src/method/formulas/initial.ts:60]: Where did this # come from?

**Review by austinborn — COMMENTED** [review]
Looks good from a technical standpoint. Biggest concern is adding sources for the magic numbers so future maintainers know how to find the values themselves. Either documentation, or maybe even the contract/tx that set the values.

File: packages/composites/dydx-rewards/test/integration/fixtures.ts (hunk 2 above)
Comment (boxhock) [reply packages/composites/dydx-rewards/test/integration/fixtures.ts:558]: It is not

File: packages/composites/dydx-rewards/src/method/formulas/initial.ts (hunk 3 above)
Comment (boxhock) [reply packages/composites/dydx-rewards/src/method/formulas/initial.ts:34]: These numbers come from internal documentation I cannot link publicly. I'll have a walkthrough of this with you and Justin next week.

File: packages/composites/dydx-rewards/src/method/formulas/initial.ts (hunk 4 above)
Comment (boxhock) [reply packages/composites/dydx-rewards/src/method/formulas/initial.ts:60]: Same as above

File: packages/composites/dydx-rewards/src/method/formulas/initial.ts (hunk 3 above)
Comment (austinborn) [reply packages/composites/dydx-rewards/src/method/formulas/initial.ts:34]: Understood, yes that would be great

**Review by austinborn — APPROVED** [review]

**Pull request merged**